src/segmented-control/src/SegmentedControl.js

```diff
@@ -1,144 +1,137 @@
-
-    /**
-     * The height of the Segmented Control.
-     */
-    height: PropTypes.number,
-
-    /**
-     * When true, the Segmented Control is disabled.
-     */
-    disabled: PropTypes.bool
+const SegmentedControl = memo(props => {
+  const {
+    value,
+    name,
+    height = 32,
+    options,
+    onChange,
+    defaultValue,
+    disabled,
+    ...rest
+  } = props
+
+  const groupName = `SegmentedControl-${radioCount}`
+  radioCount += 1
+
+  const isControlled = () => {
+    return typeof value !== 'undefined' && value !== null
   }
 
-  static defaultProps = {
-    height: 32
-  }
-
-  constructor(props, context) {
-    super(props, context)
-
-    let value = props.defaultValue
-    if (typeof value === 'undefined' || value === null) {
-      value = props.options[0].value
+  const getDefaultValue = () => {
+    if (isControlled()) {
+      return value
     }
 
-    this.state = {
-      value
-    }
-
-    this.name = `SegmentedControl-${radioCount}`
-    radioCount += 1
+    return typeof defaultValue !== 'undefined' && defaultValue !== null ? defaultValue : options[0].value
   }
 
-  isControlled = () => {
-    return typeof this.props.value !== 'undefined' && this.props.value !== null
-  }
+  const [activeValue, setActiveValue] = useState(getDefaultValue())
 
-  handleChange = value => {
-    // Save a render cycle when it's a controlled input
-    if (!this.isControlled()) {
-      this.setState({ value })
+  useEffect(() => {
+    if (isControlled() && value !== activeValue) {
+      setActiveValue(value)
     }
+  }, [value])
 
-    safeInvoke(this.props.onChange, value)
-  }
-
-  render() {
-    const {
-      value: filterOutValue, // Filter out.
-      name,
-      height,
-      options,
-      onChange,
-      defaultValue,
-      disabled,
-      ...props
-    } = this.props
-
-    // Allows it to behave like a controlled input
-    let { value } = this.state
-    if (this.isControlled()) {
-      value = this.props.value
+  const handleChange = newValue => {
+    // Save a render cycle when it's a controlled input
+    if (!isControlled()) {
+      setActiveValue(newValue)
     }
 
-    return (
-      <Box display="flex" marginRight={-1} height={height} {...props}>
-        {options.map((option, index) => (
-          <SegmentedControlRadio
-            key={option.value}
-            id={this.name + index}
-            name={name || this.name}
-            label={option.label}
-            value={String(option.value)}
-            height={height}
-            checked={value === option.value}
-            onChange={this.handleChange.bind(null, option.value)}
-            appearance="default"
-            isFirstItem={index === 0}
-            isLastItem={index === options.length - 1}
-            disabled={disabled}
-          />
-        ))}
-      </Box>
-    )
+    safeInvoke(onChange, newValue)
   }
+
+  return (
+    <Box display="flex" marginRight={-1} height={height} {...rest}>
```

Comment:
Do we want/need to support ref forwarding on the outer Box component?

Reply:
Might not be a bad idea to support. I'll add it in